generators/csharp/codegen/src/ast/__test__/Class.test.ts

```diff
@@ -0,0 +1,39 @@
+import { csharp } from "../..";
+
+describe("class", () => {
+    it("basic", async () => {
+        const clazz = csharp.class_({
+            name: "Car",
+            namespace: "Automotive",
+            access: "public",
+            primaryConstructor: {
+                parameters: [
+                    csharp.parameter({
+                        name: "make",
+                        type: csharp.Type.string()
+                    }),
+                    csharp.parameter({
+                        name: "model",
+                        type: csharp.Type.string()
+                    })
+                ],
+                superClassArguments: []
+            }
+        });
+        expect(
+            clazz.toString({
+                namespace: "",
+                allNamespaceSegments: new Set<string>(),
+                allTypeClassReferences: new Map<string, Set<string>>(),
+                rootNamespace: "",
+                customConfig: {}
+            })
+        ).toEqual(`\
```

Comment:
@noanflaherty want to update this to be `toMatchSnapshot()` ? that way we'll have a dedicated file with the generated code and itll be easier to track diffs.

Reply:
Yeah good call. I'll look for examples of `toMatchSnapshot` elsewhere.